### DISCUSSION TITLE
xoffset relative position

### DISCUSSION THREAD

**Discussion by marin-jurjevic** [davatorium/rofi#1904] [Q&A] [answered]
## Problem

Hi,

I'm using rofi in combination with polybar. When I want to give more information, I hook a rofi script on polybar icon, and clicking the icon opens a rofi popup.

Example of such script is following:

```bash
#!/bin/bash

x_mouse=$(xdotool getmouselocation | cut -d ' ' -f1 | cut -d ':' -f2) 
x_offset=$(( $x_mouse - 80 ))

# polybar height
y_offset=-25

selection=$(rofi -theme ssh_sessions -modi "sshc:~/.local/bin/ssh_sessions" -show sshc -location 7 -xoffset $x_offset -yoffset $y_offset)
```

I'm fetching current cursor position (which is where icon is position on polybar) and I pass this information to rofi so it knows where to create popup. This worked fine on first monitor, but now I moved some icons to second monitor and it seems rofi uses `x_offset` as relative to the current monitor/display. 

The issue is that `xdotool` returns absolute position of the cursor, so if my icon is at x=2100 absolute, it's actually 180 relative in rofi space (my first monitor is 1920 width). In this case, rofi receives `-xoffset  2080` and opens popup on third monitor.

## Question

1. Is there a way to use absolute offset, in complete display space. Since I'm using Xorg, from its perspective I have one giant screen.
2. Do you know for a CLI tool which can fetch dimensions of the current monitor and then I do the calculations myself? In this case I would need to know the actual order & orientation of monitors.

**Comment by DaveDavenport** [comment, marked as answer]
You can force rofi to use the first monitor, does that help if you pass absolute distance for positioning it on the 2nd?

this worked for me to position it on monitor 0 launched on any monitor.
```bash
rofi -theme-str 'window {x-offset: 3900; anchor: north west; location: north west; } configuration { monitor: 1; } '
```

**Reply by marin-jurjevic** [reply]
Great, that works. I just had to use `monitor: 0` in my configuration.
